packages/graphql-hooks/src/LocalGraphQLClient.ts

```diff
@@ -41,29 +47,38 @@ class LocalGraphQLClient extends GraphQLClient {
     // Skips all config verification from the parent class because we're mocking the client
   }
 
-  request<ResponseData = any, TGraphQLError = object, TVariables = object>(
-    operation
-  ): Promise<Result<any, TGraphQLError>> {
-    if (!this.localQueries[operation.query]) {
-      throw new Error(
-        `LocalGraphQLClient: no query match for: ${operation.query}`
-      )
-    }
-    return timeoutPromise(this.requestDelayMs)
-      .then(() =>
-        Promise.resolve(
-          this.localQueries[operation.query](
-            operation.variables,
-            operation.operationName
-          )
+  requestViaHttp<ResponseData, TGraphQLError = object, TVariables = object>(
+    operation: Operation<TVariables>,
+    options: RequestOptions = {}
+  ): Promise<Result<ResponseData, TGraphQLError>> {
+    return timeoutPromise(this.requestDelayMs).then(() => {
+      if (!operation.query || !this.localQueries[operation.query]) {
+        throw new Error(
+          `LocalGraphQLClient: no query match for: ${operation.query}`
         )
+      }
+
+      const data = this.localQueries[operation.query](
+        operation.variables,
+        operation.operationName
       )
+
+      return applyResponseReducer(options.responseReducer, data, new Response())
+    })
+  }
+
+  request<ResponseData, TGraphQLError = object, TVariables = object>(
+    operation: Operation<TVariables>,
+    options?: RequestOptions
+  ): Promise<Result<any, TGraphQLError>> {
+    return super
+      .request<ResponseData, TGraphQLError, TVariables>(operation, options)
       .then(result => {
         if (result instanceof LocalGraphQLError) {
           return { error: result }
         }
-        const { data, errors } = collectErrorsFromObject(result)
-        if (errors.length > 0) {
+        const { data, errors } = collectErrors(result)
+        if (errors && errors.length > 0) {
```

Comment:
**nitpick:** This line can be `if (errors?.length > 0) {`

Reply:
Thanks for the feedback.
I tried to use it before push the code, but I got this error from TS

![image](https://github.com/user-attachments/assets/c9c9d335-eed4-4ee3-a548-f35470dc0988)

So I preferred to stay with this version :D

Reply:
Oh okay. The current code looks good to me then. 🙂